### PR TITLE
Support all Secret keys in StorageClass

### PR DESCRIPTION
**Pull request by alonfirestein**
… secret name-namespace variables

Signed-off-by: Alon Firestein <alon.firestein@ibm.com>

### CODE REVIEW

**Comment by alonfirestein** [comment]
[Align with CSI supported names of keys of secret name and namespace](https://trello.com/c/Nz2PKAMt/164-align-with-csi-supported-names-of-keys-of-secret-name-and-namespace)

File: controllers/storageclass/storageclass_controller.go [hunk 1]
```diff
@@ -192,6 +187,34 @@ func (r *StorageClassWatcher) getFlashSystemClusterByStorageClass(sc *storagev1.
...
+	secret := &corev1.Secret{}
+	secretName, ok := sc.Parameters[util.DefaultSecretNameKey]
+	if !ok {
+		secretName, ok = sc.Parameters[util.ProvisionerSecretNameKey]
+		if !ok {
+			return *secret, fmt.Errorf("cannot find/identify secret name in StorageClass")
+		}
+	}
+	secretNamespace, ok := sc.Parameters[util.DefaultSecretNamespaceKey]
+	if !ok {
```
Comment (tyichye19) [inline controllers/storageclass/storageclass_controller.go:200]: it means that one can create a secret using default secret name, but provisioner secret namespace.
is it supported? for creating a valid PVC

File: controllers/storageclass/storageclass_controller.go [hunk 2]
```diff
@@ -192,6 +187,34 @@ func (r *StorageClassWatcher) getFlashSystemClusterByStorageClass(sc *storagev1.
 	return foundCluster, fmt.Errorf(msg)
 }
 
+func (r *StorageClassWatcher) getSecret(sc *storagev1.StorageClass) (corev1.Secret, error) {
+	secret := &corev1.Secret{}
+	secretName, ok := sc.Parameters[util.DefaultSecretNameKey]
+	if !ok {
+		secretName, ok = sc.Parameters[util.ProvisionerSecretNameKey]
+		if !ok {
+			return *secret, fmt.Errorf("cannot find/identify secret name in StorageClass")
```
Comment (tyichye19) [inline controllers/storageclass/storageclass_controller.go:196]: no need for `find/identify`, just choose one

File: controllers/storageclass/storageclass_controller.go [hunk 3]
```diff
@@ -192,6 +187,41 @@ func (r *StorageClassWatcher) getFlashSystemClusterByStorageClass(sc *storagev1.
...
+	secretName, ok := sc.Parameters[util.DefaultSecretNameKey]
+	if !ok {
+		secretName, ok = sc.Parameters[util.ProvisionerSecretNameKey]
+		if !ok {
+			return *secret, fmt.Errorf("cannot find secret name in StorageClass")
+		}
+		isSamePrefixCheck = true
+	}
+	secretNamespace, ok := sc.Parameters[util.DefaultSecretNamespaceKey]
+	if !ok && isSamePrefixCheck {
```
[the remark this answers is not in the document]
Comment (alonfirestein) [reply controllers/storageclass/storageclass_controller.go]: agreed, changed now.

File: controllers/storageclass/storageclass_controller.go [hunk 4]
```diff
@@ -192,6 +187,28 @@ func (r *StorageClassWatcher) getFlashSystemClusterByStorageClass(sc *storagev1.
...
 }
 
+func (r *StorageClassWatcher) getSecret(sc *storagev1.StorageClass) (corev1.Secret, error) {
+	secret := &corev1.Secret{}
+	secretName, secretNamespace := sc.Parameters[util.DefaultSecretNameKey], sc.Parameters[util.DefaultSecretNamespaceKey]
+	if secretName == "" || secretNamespace == "" {
+		secretName, secretNamespace = sc.Parameters[util.ProvisionerSecretNameKey], sc.Parameters[util.ProvisionerSecretNamespaceKey]
+		if secretName == "" || secretNamespace == "" {
+			r.Log.Error(nil, "failed to find secret name or namespace in StorageClass")
+			return *secret, fmt.Errorf("failed to find secret name or namespace in StorageClass")
```
Comment (tyichye19) [inline controllers/storageclass/storageclass_controller.go:197]: please extract the error message into variable and use it for both `Log.Error` and `fmt.Error`

**Review by tyichye19 — APPROVED** [review]
LGTM

**Pull request merged**